framework/packages/abstract-client/src/account.rs

```diff
@@ -317,25 +336,95 @@ impl<'a, Chain: CwEnv> AccountBuilder<'a, Chain> {
             install_modules,
             account_id: self.expected_local_account_id,
         };
-        let abstract_account = if let Some(owner_account) = self.owner_account {
-            owner_account
-                .abstr_account
-                .create_sub_account(account_details, Some(&funds))?
-        } else {
-            self.abstr.account_factory.create_new_account(
+        let abstract_account = match self.owner_account {
+            None => self.abstr.account_factory.create_new_account(
                 account_details,
                 ownership,
                 Some(&funds),
-            )?
+            )?,
+            Some(OwnerAccount::CurrentChain(owner_account)) => owner_account
+                .abstr_account
+                .create_sub_account(account_details, Some(&funds))?,
+            Some(OwnerAccount::RemoteChain(_)) => {
+                return Err(AbstractClientError::RemoteAccountOnBuild {})
+            }
         };
         Ok(Account::new(abstract_account, self.install_on_sub_account))
     }
+
+    /// Builds the [`RemoteAccount`].
```

Comment:
If we have a fn for building a remote then why is there a `remote_account` fn? Shouldn't the regular build fn recognise that a user is trying to create a remote account and then build for that remote account?

Reply:
> If we have a fn for building a remote then why is there a `remote_account` fn?

Similar to `sub_account` where we pass "owner_account" but it's even more useful in this case because then we can use 

> Shouldn't the regular build fn recognise that a user is trying to create a remote account and then build for that remote account?

It can, but then we have to either add generic to the `AccountBuilder` or second builder for one method difference (perhaps you have a better idea?), felt like separate method would be nicer.

Reply:
> Hmmm does it even make sense to have a builder for remote accounts? We can't create remote accounts without having a local account so maybe we should get a seperate builder from an account to create a remote account?
> 
> I just feel like it will be confusing for users to use this builder method to create remote accounts as they can easily mis-use it.
> 
> What if we make this method private and create a seperate `RemoteAccountBuilder` that wraps this and that is constructed from an existing Account so its governance details are set correctly?

Sure, let's have separate RemoteAccountBuilder!